### DISCUSSION TITLE
Re-training BERTopic Model

### DISCUSSION THREAD

**Discussion by dherysetiaone** [MaartenGr/BERTopic#1735] [Q&A] [answered]
Hii,

Currently i have a running BERTopic model. The model works fine for 3 months until recently there are new patterns that not yet to be learn by the existing categories. I want this existing categories to learn this new pattern without adding or removing any current categories. So i re-trained this model by just fit_transform the additional data train to the current model. But after i do that, the topic name no longer represent the said topic number. For example: before the changes, topic number 8 is about cars, but after i do that, the said topic number 8 is now about animal. Is there any way to make the topic name consistent? or there is other way that i should try for retraining my model.

Thank you.

**Comment by MaartenGr** [comment, marked as answer]
That is exactly what `.merge_models` is for: https://maartengr.github.io/BERTopic/getting_started/merge/merge.html

You can train a new topic model and merge it with your existing model.

**Reply by dherysetiaone** [reply]
Thank you so much for the quick answer! Definitely will try this out.
